tools/java/org/unicode/cldr/util/DtdData.java

```diff
@@ -652,22 +653,41 @@ public void handleEndDtd() {
     }
 
     /**
-     * Normal version of DtdData
      * Note that it always gets the trunk version
+     * @deprecated depends on static config, use {@link DtdData#getInstance(DtdType, File)} instead
      */
     public static DtdData getInstance(DtdType type) {
-        return CACHE.get(type);
+        return getInstance(type, CLDRConfig.getInstance().getCldrBaseDirectory());
     }
-
+    
     /**
      * Special form using version, used only by tests, etc.
      */
     public static DtdData getInstance(DtdType type, String version) {
-        DtdData simpleHandler = new DtdData(type, version);
-        XMLFileReader xfr = new XMLFileReader().setHandler(simpleHandler);
         File directory = version == null ? CLDRConfig.getInstance().getCldrBaseDirectory()
             : new File(CLDRPaths.ARCHIVE_DIRECTORY + "/cldr-" + version);
 
+        return getInstance(type, version, directory);
+    }
+
+    private static final ConcurrentMap<Pair<DtdType, File>, DtdData> CACHE = new ConcurrentHashMap<>();
+
+    /**
+     * Normal version of DtdData
+     * Get a DtdData, given the CLDR root directory.
+     * @param type which DtdType to return
+     * @param directory the CLDR Root directory, which contains the "common" directory.
+     * @return
+     */
+    public static DtdData getInstance(DtdType type, File directory) {
+        Pair<DtdType, File> key = new Pair<>(type, directory);
```

Comment:
You can obviously just "return" directly now (no "data" variable), and if you wanted to inline the key, that'll make a single statement (but probably split on 2 lines, so maybe not worth it).

Reply:
thanks. I split it because of the line complexity with the lambda